tests/rptest/tests/read_replica_e2e_test.py

```diff
@@ -170,12 +140,91 @@ def s3_has_all_data():
             timeout_sec=
             30,  #should be uploaded since cloud_storage_segment_max_upload_interval_sec=5
             backoff_sec=5,
-            err_msg=
-            f"Not all data is uploaded to S3 bucket, is S3 bucket: {list(self.redpanda._s3client.list_objects(self.s3_bucket_name))}"
-        )
+            err_msg="Not all data is uploaded to S3 bucket: " +
+            f"{list(s3_client.list_objects(self.s3_bucket_name))}")
 
         # Create read replica topic, consume from it and validate
         self.start_second_cluster()
         self.create_read_replica_topic()
-        self.start_consumer()
+
+    def _bucket_usage(self) -> BucketUsage:
+        assert self.redpanda and self.redpanda.s3_client
+        keys: set[str] = set()
+        s3 = self.redpanda.s3_client
+        num_objects: int = 0
+        total_bytes: int = 0
+        for o in s3.list_objects(self.s3_bucket_name):
+            num_objects += 1
+            total_bytes += o.ContentLength
+            keys.add(o.Key)
+        self.redpanda.logger.info(
+            f"bucket usage {num_objects} objects, " +
+            f"{total_bytes} bytes for {self.s3_bucket_name}")
+        return BucketUsage(num_objects, total_bytes, keys)
+
+    def _bucket_delta(self, bu1: BucketUsage,
+                      bu2: BucketUsage) -> Optional[BucketUsage]:
+        """ Return None if both bucket usage reports are equal.
+            Otherwise, return deltas. """
+        obj_delta: int = 0
+        bytes_delta: int = 0
```

Comment:
q: why do we need to initialize these separately?

Reply:
We don't. This is leftover from a refactor I did. I will simplify this, thanks.

Reply:
fixed, thanks.